wallet/src/account/mod.rs

```diff
@@ -2514,6 +2582,67 @@ fn group_preselected_inputs(
     Ok(preselected_inputs)
 }
 
+fn handle_fill_order_op(
+    order_id: OrderId,
+    fill_amount_in_ask_currency: Amount,
+    order_info: Option<&BTreeMap<OrderId, &RpcOrderInfo>>,
+    fee: Amount,
+    update_preselected_inputs: &mut impl FnMut(Currency, Amount, Amount, Amount) -> WalletResult<()>,
+) -> WalletResult<()> {
+    let order_info = order_info
+        .as_ref()
+        .and_then(|info| info.get(&order_id))
+        .ok_or(WalletError::OrderInfoMissing(order_id))?;
+
+    let filled_amount = orders_accounting::calculate_filled_amount(
+        order_info.ask_balance,
```

Comment:
Probably it's better to use initially_asked/initially_given here and/or assert that they equal ask_balance/give_balance?